### PR TITLE
Refactor/stt

### PR DESCRIPTION
**Pull request by khh0303**
update stt compile/runtime code

### CODE REVIEW

**Comment by khh0303** [comment]
@mobilint-review

File: runtime/python/stt/prepare_model.py [hunk 1]
```diff
@@ -1,167 +1,116 @@
...
+
+def patch_config(output_folder: str, encoder_mxq: str, decoder_mxq: str) -> None:
+    """Point config.json's mxq paths at the copied files; keep the repo's core allocation."""
     config_path = os.path.join(output_folder, "config.json")
-    print(f"Saving configuration to {config_path}")
+    with open(config_path) as f:
+        config = json.load(f)
+
+    config["encoder_mxq_path"] = encoder_mxq
+    config["decoder_mxq_path"] = decoder_mxq
```
[the remark this answers is not in the document]
Comment (khh0303) [reply runtime/python/stt/prepare_model.py:63]: 올바른 지적입니다. 하지만 아직 regulus-rb에 대한 huggingface repo가 준비가 안되었습니다. 현재로써는 aries-rb에서만 동작을 보장하고 있습니다. 추후 허깅페이스에 regulus-rb 레포 생성후 분기를 추가하겠습니다.

---  
Good point. However, the Hugging Face repository for regulus-rb is not available yet. At the moment, we only support aries-rb. We will update the code after the repository for regulus-rb is ready.

File: compilation/stt/generate_calibration.py [hunk 2]
```diff
@@ -30,189 +47,174 @@
...
-    Uses whisper-small to generate transcriptions and translations on-the-fly,
-    then creates decoder input embeddings (token + positional) as calibration data.
+    Returns None if generation produced no tokens.
+    """
+    decoder = model.model.decoder
+    features = processor(audio_chunk, sampling_rate=16000, return_tensors="pt").input_features.to(model.device)
+    with torch.no_grad():
+        encoder_hidden = model.model.encoder(features).last_hidden_state
+        tokens = model.generate(features, language=language, task=task)[0]
+        if tokens.numel() == 0:
```
[the remark this answers is not in the document]
Comment (khh0303) [reply compilation/stt/generate_calibration.py:128]: 확인 결과, 현재 생성된 캘리브레이션 데이터에는 특수 토큰만으로 구성된 샘플이 없었습니다.

또한 실험 결과, 이전의 4토큰 이하 시퀀스를 제외하는 로직은 오히려 양자화 정확도를 저하시켰습니다. 현재 사용 중인 `tokens.numel() == 0` 조건이 가장 높은 양자화 정확도를 보였습니다.

현재 조건으로 생성된 캘리브레이션 데이터에는 하나의 실제 콘텐츠 토큰만 포함된 유효한 샘플도 존재합니다. 이러한 짧은 유효 샘플을 유지한 것이 양자화 성능 향상에 기여했을 가능성이 있다고 판단합니다.

따라서 현재 구현을 유지하고, 제안해 주신 변경 사항은 반영하지 않겠습니다.

---  

We verified that none of the calibration samples generated by the current implementation contain only special tokens.

Our experiments also showed that the previous logic of discarding sequences with four or fewer tokens actually reduced quantization accuracy. The current `tokens.numel() == 0` condition produced the best quantization accuracy.

The calibration dataset generated with the current condition includes valid samples containing only a single content token. We believe that retaining these short but valid samples may have contributed to the improved quantization performance.

Therefore, we will keep the current implementation and will not apply the suggested change.

**Comment by khh0303** [comment]
@mobilint-review

File: compilation/stt/generate_calibration.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (khh0303) [reply compilation/stt/generate_calibration.py:128]: 실제 저장된 캘리브레이션 데이터를 검증한 결과, 특수 토큰이나 제어 토큰만으로 구성된 샘플은 존재하지 않았습니다. `transformers` 4.50 환경에서는 `generate()`가 실제 콘텐츠 토큰만 반환하는 것을 확인했습니다.
따라서 제안해 주신 "특수 토큰만 포함된 출력을 제외하는" 필터는 현재 생성되는 데이터셋에서는 아무런 샘플도 제거하지 않습니다.
또한, 실제 콘텐츠 토큰 수가 4개 이하인 저-내용 샘플(120개)을 제외하는 실험도 수행해 보았으나, 오히려 WER가 악화되어 해당 변경은 적용하지 않았습니다.


We verified the generated calibration dataset and found that none of the saved samples contain only special or control tokens. With `transformers` 4.50, `generate()` returns only content tokens in our setup.
Therefore, the proposed filter to discard outputs containing only special/control tokens would not remove any samples from the current calibration dataset.
We also evaluated filtering out low-content samples (those containing four or fewer content tokens, 120 samples in total). However, this resulted in worse WER, so we decided to keep the current implementation.

**Comment by khh0303** [comment]
@mobilint-review

File: compilation/stt/generate_calibration.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (khh0303) [reply compilation/stt/generate_calibration.py:128]: 실제 저장된 캘리브레이션 데이터를 검증한 결과, 특수 토큰이나 제어 토큰만으로 구성된 샘플은 존재하지 않았습니다. transformers 4.50 환경에서는 generate()가 실제 콘텐츠 토큰만 반환하는 것을 확인했습니다.
따라서 제안해 주신 "특수 토큰만 포함된 출력을 제외하는" 필터는 현재 생성되는 데이터셋에서는 아무런 샘플도 제거하지 않습니다.
또한, 실제 콘텐츠 토큰 수가 4개 이하인 저-내용 샘플(120개)을 제외하는 실험도 수행해 보았으나, 오히려 WER가 악화되어 해당 변경은 적용하지 않았습니다.

We verified the generated calibration dataset and found that none of the saved samples contain only special or control tokens. With transformers 4.50, generate() returns only content tokens in our setup.
Therefore, the proposed filter to discard outputs containing only special/control tokens would not remove any samples from the current calibration dataset.
We also evaluated filtering out low-content samples (those containing four or fewer content tokens, 120 samples in total). However, this resulted in worse WER, so we decided to keep the current implementation.

File: runtime/python/stt/README.md [hunk 3]
```diff
@@ -35,18 +35,19 @@ The prepared model folder stores the encoder MXQ, decoder MXQ, embedding weights
...
+    --force
 ```
 
 This script:
 
-- Copies the compiled encoder and decoder MXQ files
-- Downloads the base Whisper configuration
-- Extracts decoder embedding weights into `model.safetensors`
-- Writes `config.json` with default NPU core allocation
+- Downloads the Hugging Face repo via `huggingface_hub.snapshot_download` (self-contained `config.json`, proxy classes, tokenizer, generation config, and `model.safetensors`), skipping only the repo's own `.mxq`
```
[the remark this answers is not in the document]
Comment (khh0303) [reply runtime/python/stt/README.md:46]: 리뷰 감사합니다.

해당 가이드 문서는 레포지토리의 전체 아키텍처와 모델 준비 과정에 대한 high-level 개요를 제공하기 위한 문서이며, 세부 구현 변경 사항을 모두 반영하는 것을 목적으로 하지 않습니다.

이번 PR의 변경은 STT 모델 준비 과정의 내부 구현만 변경할 뿐, 전체 워크플로우나 레포지토리의 설계 구조에는 영향을 주지 않습니다. 따라서 이번 PR에서 공용 가이드 문서를 수정할 필요는 없다고 판단하였습니다.
  
Thank you for the review.

These guide documents are intended to provide a high-level overview of the repository architecture and the model preparation workflow, rather than to document every implementation detail.

The changes in this PR only affect the internal implementation of the STT model preparation process and do not change the overall workflow or repository architecture. Therefore, we do not believe that updating the shared guide documents is necessary as part of this PR.

**Review by parkjinman98 — APPROVED** [review]
